inconspiquous/dialects/qu.py

```diff
@@ -51,6 +67,37 @@ def num_qubits(self) -> int:
         return 1
 
 
+@irdl_op_definition
+class FromBitsOp(IRDLOperation):
+    name = "qu.from_bits"
+    qubits = var_operand_def(BitType)
+    reg = result_def(QReg)
```

Comment:
The xDSL constraint system allows us to constrain the various integers here, so we can do something like:
```suggestion
    _I : ClassVar = IntVarConstraint("I", AnyInt())
    
    qubits = var_operand_def(RangeOf(BitType, length=_I))
    reg = result_def(ParamAttrConstraint(QReg, (IntAttrConstraint(_I),))
```
an even cleaner way to do this would be to define a `constr` method on `QReg` (see https://github.com/xdslproject/xdsl/blob/8a61f4d0f7a49452c75363e7a8f9b46d5eaaebca/xdsl/dialects/builtin.py#L820 for an example)